app/services/process_stripe_payment.rb

```diff
@@ -0,0 +1,25 @@
+class ProcessStripePayment
+  attr_reader :controller, :amount, :email, :token
+  
+  def initialize(controller, amount, email, token)
+    @controller = controller
+    @amount = amount
+    @email = email
+    @token = token
+  end
+  
+  def charge_card
+    begin
+      Stripe.api_key = ENV['STRIPE_SECRET_KEY']
+      Stripe::Charge.create(
+        amount: amount,
+        currency: "gbp",
+        source: token,
+        description: "Charge for #{email}"
+      ) 
+    rescue Stripe::CardError => e
+      controller.flash[:danger] = e.message
```

Comment:
This shouldn't be here really. Not that it wouldn't work but because setting flashes, working with params hash, redirecting etc this is a concern of Controller and it should stay and be handled there. This way you moved it away. Also similar parallel is not to move params hash to model layer etc. You will hear Kevin speaking about it in some solution week 7.